### PR TITLE
Lesson 2.1

### PR DESCRIPTION
**Pull request by secch97**
- Converted GuessControl class component into a functional component.
- Converted NumberGuessingGame class component into a functional component.
- Fixed a bug: Once the game was over, the user was still able to submit more guesses... I disabled the input and button once the game is finished

### CODE REVIEW

File: src/GuessControl.js [hunk 1]
```diff
@@ -1,47 +1,32 @@
...
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
+  const handleInputChange = (event) => {
+    setCurrentGuess(event.target.value);
   }
 
-  handleInputChange(event) {
-    this.setState({ currentGuess: event.target.value });
-  }
+  const onSubmitGuess = () => {
+    onGuess(+currentGuess);
```
Comment (viniciuspietscher) [inline src/GuessControl.js:12]: That's very cool! I didn't know we could convert to a number like that! 💯

File: src/GuessControl.js
Comment (viniciuspietscher) [inline src/GuessControl.js:1]: Great job! 🥇

File: src/Button.js [hunk 2]
```diff
@@ -4,9 +4,9 @@ const buttonStyle = {
   minHeight: "2em",
 };
 
-export default function Button({ onClick, children }) {
+export default function Button({ onClick, disabled, children }) {
   return (
-    <button style={buttonStyle} onClick={onClick}>
+    <button style={buttonStyle} onClick={onClick} disabled={disabled}>
```
Comment (viniciuspietscher) [inline src/Button.js:9]: 💯 🥇

File: src/NumberGuessingGame.js [hunk 3]
```diff
@@ -13,65 +13,54 @@ function getRandomNumber() {
...
-            guess={this.state.latestGuess}
-            numberToGuess={this.state.numberToGuess}
-            numberOfGuesses={this.state.numberOfGuesses}
+  return (
+    <div>
+      <h2>I'm thinking of a number from 1 to 100.</h2>
+      <h2>
+        Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
+      </h2>
+      <GuessControl onGuess={handleGuess} disabled={isGameOver}/>
```
Comment (viniciuspietscher) [inline src/NumberGuessingGame.js:44]: That's very cool! I don't think I even noticed this when I worked on this 😅 
🥇 💯

File: src/NumberGuessingGame.js
Comment (viniciuspietscher) [inline src/NumberGuessingGame.js:1]: Great job!!! 🥇 💯

**Review by viniciuspietscher — APPROVED** [review]
Great job Saul! 🥇 💯

File: src/GuessControl.js (hunk 1 above)
Comment (secch97) [reply src/GuessControl.js:12]: Yeah! It's pretty fast to do it like this haha

File: src/NumberGuessingGame.js (hunk 3 above)
Comment (secch97) [reply src/NumberGuessingGame.js:44]: Thanks haha, I always like to test everything 😅

File: src/NumberGuessingGame.js
Comment (secch97) [reply src/NumberGuessingGame.js:1]: Thank you Vini!

**Pull request merged**